### DISCUSSION TITLE
Using custom users table and provider

### DISCUSSION THREAD

**Discussion by ilicmarko** [statamic/cms#8125] [Troubleshooting] [answered]
Something strange is happening and I can't wrap my head around it. 😭 

I have been trying to setup Statamic to store users in the database without luck. This is an existing Laravel app and I am trying to use a separate users table from the existing one. Here is my `auth.php`.

Note I have successfully created the Users and other tables (as I am using the eloquent driver for storing content in the DB).

```php
'guards' => [
        //..
        'statamic' => [
            'driver' => 'session',
            'provider' => 'statamic',
        ]
    ],

    'providers' => [
        //...
        'statamic' => [
            'driver' => 'statamic',
        ],
    ],

    'passwords' => [
       //...
        'activations' => [
            'provider' => 'statamic',
            'table' => 'password_activations',
            'expire' => 4320,
            'throttle' => 60,
        ],
    ],
```

Here is the `users.php`

```php
[
    'repository' => 'eloquent',

    'repositories' => [

        'file' => [
            'driver' => 'file',
            'paths' => [
                'users' => base_path('users'),
                'roles' => resource_path('users/roles.yaml'),
                'groups' => resource_path('users/groups.yaml'),
            ],
        ],

        'eloquent' => [
            'driver' => 'eloquent',
        ],

    ],

  'database' => env('STATAMIC_ELOQUENT_CONNECTION'), // Maybe this is the problem?

    'tables' => [
        'users' => 'users',
        'role_user' => 'role_user',
        'group_user' => 'group_user',
    ],

    'guards' => [
        'cp' => 'statamic',
        'web' => 'statamic',
    ],
]
```

The `STATAMIC_ELOQUENT_CONNECTION` is the separate database connection I use for Statamic, which I configured in the `database.php`.

Finally, this is the error I get when I try to login or create a new user, config modal is null:

```
call_user_func_array(): Argument #1 ($callback) must be a valid callback, first array member is not a valid class name or object
```

![image](https://github.com/statamic/cms/assets/2827249/b7389538-1c35-4cbc-bb09-7fd82526ee67)

Any ideas?

**Comment by BeauvR** [comment, marked as answer]
Hi, I encountered the same issue and found a workaround.

I included the following guard in the `auth.php`:
```
        'statamic' => [
            'driver' => 'session',
            'provider' => 'statamic',
        ],
```

I also added the subsequent provider to the `auth.php`:
```
        'statamic' => [
            'driver' => 'eloquent',
            'model' => App\Models\StatamicUser::class,
        ],
```

Furthermore, I created a custom model for the table which extends the `Illuminate\Foundation\Auth\User`.

The only modification I made was to the CP authentication guard in the `config/statamic/users.php`, setting it to `statamic`.

I hope this solution will work for others as well.
